cmd/snap/cmd_run.go

```diff
@@ -533,8 +533,8 @@ func runCmdUnderStrace(origCmd, env []string) error {
 	if err := gcmd.Start(); err != nil {
 		return err
 	}
-	err = gcmd.Wait()
 	<-filterDone
+	err = gcmd.Wait()
```

Comment:
In general a bool channel is nicer to read for that use case (filterDone <- true). It's also a nice practice to have the sending into a defer, so that the channel is guaranteed to unblock no matter the exit path, and for the same reason but on the opposite end, it's also nice to use a buffered channel, so that if the outer goroutine doesn't attempt to receive for whatever reason, the internal one still terminates.

Those suggestions are unlikely to solve anything in the current code, but might avoid future issues.

Reply:
Thanks, that is excellent feedback!